### ISSUE TITLE
Wrap user-defined widgets

### ISSUE THREAD

**Issue by danielshahaf** [zsh-users/zsh-syntax-highlighting#203]
`_zsh_highlight_bind_widgets` uses the following code to decide which widgets to wrap:

```zsh
  for cur_widget in ${${(f)"$(builtin zle -la)"}:#(.*|_*|orig-*|run-help|which-command|beep|yank-pop)}; do
    case $widgets[$cur_widget] in

      # Already rebound event: do nothing.
      user:$cur_widget|user:_zsh_highlight_widget_*);;
```

User-defined widgets, such as `edit-command-line` or even a trivial `f() {}; zle -N f` (which both fall into the first branch of that `case` pattern), should be wrapped to recompute highlighting like other widgets.

**Comment by danielshahaf** [comment]
The redrawhook work (https://github.com/zsh-users/zsh-syntax-highlighting/issues/154#issuecomment-144541575) solves the problem.

**Comment by danielshahaf** [comment]
Right now, the code looks like this:

https://github.com/zsh-users/zsh-syntax-highlighting/blob/82f5eb17ecee1f18b3ce3d8a93761a51235cab78/zsh-syntax-highlighting.zsh#L206-L212

```zsh
  206         # Already rebound event: do nothing.
  207         user:$cur_widget|user:_zsh_highlight_widget_*);;
  208             
  209         # User defined widget: override and rebind old one with prefix "orig-".
  210         user:*) eval "zle -N orig-$cur_widget ${widgets[$cur_widget]#*:}; \
  211                       _zsh_highlight_widget_$cur_widget() { _zsh_highlight_call_widget orig-$cur_widget \"\$@\" }; \
  212                       zle -N $cur_widget _zsh_highlight_widget_$cur_widget";;
```

Why does it behave differently for `zle -N foo foo` (takes the `user:$cur_widget` case) and `zle -N foo bar` (takes the `user:*` case)?

**Comment by danielshahaf** [comment]
Fixed some time ago in 487b122c480d00befc3f80a8cc2e53ad99593b2c.
